Add enabling default instructions flag for mobile

tugsbayasgalan opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#57778 Add enabling default instructions flag for mobile**

Differential Revision: [D28268997](https://our.internmc.facebook.com/intern/diff/D28268997)

tugsbayasgalan committed:
Add enabling default instructions flag for mobile

[ghstack-poisoned]

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28268997).

tugsbayasgalan committed:
Update on "Add enabling default instructions flag for mobile"


Differential Revision: [D28268997](https://our.internmc.facebook.com/intern/diff/D28268997)

[ghstack-poisoned]

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28268997).

tugsbayasgalan committed:
Update on "Add enabling default instructions flag for mobile"


Differential Revision: [D28268997](https://our.internmc.facebook.com/intern/diff/D28268997)

[ghstack-poisoned]

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28268997).

tugsbayasgalan committed:
Update on "Add enabling default instructions flag for mobile"


Differential Revision: [D28268997](https://our.internmc.facebook.com/intern/diff/D28268997)

[ghstack-poisoned]

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28268997).

tugsbayasgalan committed:
Update on "Add enabling default instructions flag for mobile"


Differential Revision: [D28268997](https://our.internmc.facebook.com/intern/diff/D28268997)

[ghstack-poisoned]

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28268997).

tugsbayasgalan committed:
Update on "Add enabling default instructions flag for mobile"


Differential Revision: [D28268997](https://our.internmc.facebook.com/intern/diff/D28268997)

[ghstack-poisoned]

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28268997).

tugsbayasgalan committed:
Update on "Add enabling default instructions flag for mobile"


Differential Revision: [D28268997](https://our.internmc.facebook.com/intern/diff/D28268997)

[ghstack-poisoned]

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28268997).

tugsbayasgalan committed:
Update on "Add enabling default instructions flag for mobile"


Differential Revision: [D28268997](https://our.internmc.facebook.com/intern/diff/D28268997)

[ghstack-poisoned]

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28268997).

tugsbayasgalan committed:
Update on "Add enabling default instructions flag for mobile"


Differential Revision: [D28268997](https://our.internmc.facebook.com/intern/diff/D28268997)

[ghstack-poisoned]

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28268997).

gmagogsfm commented on torch/csrc/jit/serialization/export_module.cpp:
```diff
@@ -69,7 +69,7 @@ std::pair<IValue, IValue> getFunctionTuple(
   Inline(*graph);
 
-  torch::jit::MobileCode code(graph, func.name());
+  torch::jit::MobileCode code(graph, func.name(), true);
```
```suggestion
  torch::jit::MobileCode code(graph, func.name(), /*emit_default_input_instructions=*/true);
```

gmagogsfm commented on torch/csrc/jit/runtime/interpreter.h:
```diff
@@ -81,6 +81,7 @@ struct TORCH_API MobileCode : Code {
   explicit MobileCode(
       const std::shared_ptr<Graph>& graph,
       std::string function_name,
+      bool emit_default_input_instructions = false,
```
Maybe the default value should be `True`. When we decide to enable FC for mobile, we can flip value used on line 72 of export_module.cpp to `False`. Eventually, this flag can be removed.

tugsbayasgalan committed:
Update on "Add enabling default instructions flag for mobile"


Differential Revision: [D28268997](https://our.internmc.facebook.com/intern/diff/D28268997)

[ghstack-poisoned]

iseeyuan reviewed (approved):
LGTM. Thanks!
